### ISSUE TITLE
Give suggestions for new algorithm implementations.

### ISSUE THREAD

**Issue by Ashrockzzz2003** [Ashrockzzz2003/Data_Structures_and_Algorithms#8]
Share your ideas for new algorithm implementations as a reply to this issue. Open for multiple contributions. A new issue will be created if we accept a suggestion for you to work on code.

**Comment by Leela0o5** [comment]
Hi, I propose implementing the Counting Sort , which has a time complexity of O(n+range) where n is the number of elements in the array. It is useful for small-range data and the order of the duplicate elements is not disturbed(stable sorting algorithm).

**Comment by Ashrockzzz2003** [comment]
@Leela0o5 
Thanks for the suggestion. Will create a task for this and assign to you. You can work on it.

**Comment by adofm** [comment]
Hi,I would like to implement Tim sort which is derived from both merge sort and insertion sort its the default sorting algorithm for python and java i would like do implement this in cpp.
Average Time Complexity  - O(nlog(n))
        Space Complexity - O(n)

**Comment by akshayks13** [comment]
Hi , i would like to add the Djikstra's , Bellman Ford and also Floyd Warshall Algorithm in python

**Comment by Ashrockzzz2003** [comment]
Hey @adofm 
You can go ahead. Will create an issue for the same. Reply there to get it assigned.
Thanks for the idea!

**Comment by Ashrockzzz2003** [comment]
> Hi , i would like to add the Djikstra's , Bellman Ford and also Floyd Warshall Algorithm in python

Hi @akshayks13, the implementation's [already exist in a different file](https://github.com/Ashrockzzz2003/Data_Structures_and_Algorithms/blob/main/non_linear/graphs/text_book_based/shortest_paths/shortest_paths.py). You can split them up and put em inside the created folders. Will create an issue for the same.

**Comment by sanj4git** [comment]
Hey there!, I'd love to add Prim's Algorithm and Kruscal's Algorithm for which helps in identifying Minimum Spanning Trees (MSTs) in CPP

**Comment by Ashrockzzz2003** [comment]
They already exist @sanj4git https://github.com/Ashrockzzz2003/Data_Structures_and_Algorithms/blob/main/non_linear/graphs/minimum_spanning_tree/kruskal/MST_Kruskals.cpp

**Comment by Keerthivasan-Venkitajalam** [comment]
I plan to add the `M-Coloring Problem` and `Hamiltonian Cycle` algorithms to the `backtracking` directory, including implementations in both Python and C++. Let me know your thoughts.

**Comment by Ashrockzzz2003** [comment]
@Keerthivasan-Venkitajalam 
That's a great idea. Create an issue with a detailed proposal describing clearly the problem you're solving. Will assign you after taht!

**Comment by sanj4git** [comment]
> They already exist @sanj4git https://github.com/Ashrockzzz2003/Data_Structures_and_Algorithms/blob/main/non_linear/graphs/minimum_spanning_tree/kruskal/MST_Kruskals.cpp

Sorry about that! 
Can I rather contribute an implementation of AVL (Adelson Velsky Landis) Trees, which is self balancing in nature?

**Comment by Ashrockzzz2003** [comment]
Yes @sanj4git 
Create an issue with a detailed proposal of the same mentioning the languages you're planning to implement them in. Will assign to you there!

**Comment by Nandgopal-R** [comment]
Hi, I would like to implement multithreaded merge sort in go

**Comment by MithileshwaranS** [comment]
I would like to add bfs and dfs alogirthms in java

**Comment by Ashrockzzz2003** [comment]
> Hi, I would like to implement multithreaded mergesort in go

Hi @Nandgopal-R 
Approved. You can create an issue for the same. Will assign it to you! Love the idea!

**Comment by Ashrockzzz2003** [comment]
> I would like to add bfs and dfs alogirthms in java

Hi @MithileshwaranS  Done. You can create an issue for the same. I'll assign it to you

**Comment by hirux06** [comment]
Heyy!! I would like to add the backtracking type problems in java: All permutations, N queens, all subsets. (in java) - @Ashrockzzz2003

**Comment by Praneeth1-O-1** [comment]
Hey @Ashrockzzz2003 , noticed that there isn't a heap sort algo present, would like to work on that!

**Comment by Praneeth1-O-1** [comment]
Hey @Ashrockzzz2003 , noticed that there wasn't  a shell sort , I would like to implement a shell sort algorithm in java.

**Comment by Ashrockzzz2003** [comment]
> Heyy!! I would like to add the backtracking type problems in java: All permutations, N queens, all subsets. (in java) - @Ashrockzzz2003

Create an issue for the same if it doesn't exist already.

**Comment by Praneeth1-O-1** [comment]
> > Hey @Ashrockzzz2003 , noticed that there wasn't a shell sort , I would like to implement a shell sort algorithm in java.
> 
> Done @Praneeth1-O-1 Create an issue for the same if it doesn't exist already.

yeah @Ashrockzzz2003 i've done that , can you assign the issue to me?

**Comment by priyansh-narang2308** [comment]
Can I implement **Ford-Fulkerson Algorithm** which is used to find the maximum flow in a flow network.
It used  (DFS) to find paths from the source to the sink in a residual graph.
Its quite useful

**Comment by adithya-menon-r** [comment]
Hey @Ashrockzzz2003, I would like to implement the Radix Sorting Algorithm in cpp.

**Comment by Ashrockzzz2003** [comment]
Hi @adithya-menon-r
You can create an issue for the same, will assign it to you.

**Comment by Ashrockzzz2003** [comment]
> Can I implement **Ford-Fulkerson Algorithm** which is used to find the maximum flow in a flow network. It used (DFS) to find paths from the source to the sink in a residual graph. Its quite useful

Hi @priyansh-narang2308 
Come up with a proposal clearly describing the problem and the approaches you're considering. Put them in a new issue. We'll see what can be done.

**Comment by ay05h** [comment]
Hi @Ashrockzzz2003
I propose implementing the IntroSort (short for introspective sort), Commonly used in C++'s Standard Template Library (STL) for its balance between speed and reliability. I would like do implement this in Java and Python.

**Comment by Ashrockzzz2003** [comment]
Create an issue for it @ay05h. Will assign to you!
